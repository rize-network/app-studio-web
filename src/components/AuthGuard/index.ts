export { AuthGuard, withAuthGuard, RouteAuthGuard, AuthStatus } from './AuthGuard';
