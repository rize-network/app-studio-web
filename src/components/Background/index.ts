export { Background } from './Background';
export type {
  BackgroundProps,
  AuroraBackgroundProps,
  MeteorsProps,
  WallProps,
  ParticlesProps,
  GridProps,
  RipplesProps,
  BackgroundImageProps,
  BackgroundVideoProps,
  BackgroundGradientProps,
  BackgroundStyles,
} from './Background/Background.props';
export type {
  BackgroundEffect,
  AuroraDirection,
  MeteorTiming,
  BorderAnimationStyles,
  StrokeAnimationStyles,
  BackgroundContextType,
  BackgroundSize,
  BackgroundPosition,
  BackgroundRepeat,
  BlendMode,
} from './Background/Background.type';
