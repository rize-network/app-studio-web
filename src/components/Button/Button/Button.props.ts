import React, { CSSProperties } from 'react';
import { Shadow } from 'app-studio';

import { Elevation } from '../../../utils/elevation';

import {
  IconPosition,
  Shape,
  Size,
  Variant,
  LoaderPosition,
  Effects,
} from './Button.type';
import { LoaderProps } from '../../Loader/Loader/Loader.props';

/**
 * Represents the properties for the Button component.
 */
export interface ButtonProps {
  /**
   * The content to be rendered inside the button.
   */
  children?: React.ReactNode;
  /**
   * Sets the background color of the button.
   */
  colorScheme?: string;
  /**
   * Specifies the external URL used when the variant is set to "link".
   */
  externalHref?: string;
  /**
   * Indicates whether the button is in a loading state.
   */
  isLoading?: boolean;
  /**
   * Indicates whether the button is in a loading state.
   */
  loaderProps?: LoaderProps;
  /**
   * Specifies the position of the icon within the button.
   */
  loaderPosition?: LoaderPosition;
  /**
   * The icon component rendered within the button.
   */
  icon?: React.ReactNode;
  /**
   * Specifies the position of the icon within the button.
   */
  iconPosition?: IconPosition;
  /**
   * Disables the button if set to true.
   */
  isDisabled?: boolean;
  /**
   * Adjusts the width of the button to take available space.
   */
  isFilled?: boolean;
  /**
   * Adjusts the width of the button based on content size.
   */
  isAuto?: boolean;
  /**
   * Makes the icon container shape rounded if set to true.
   */
  isIconRounded?: boolean;
  /**
   * The event handler called when the button is clicked or pressed.
   */
  onClick?: Function;
  /**
   * Specifies the size of the button's text and padding.
   */
  size?: Size;
  /**
   * Sets a shadow effect applied to the button.
   */
  shadow?: Shadow | Elevation | CSSProperties;
  /**
   * Specifies the shape of the button's corners.
   */
  shape?: Shape;
  /**
   * Descriptive label for accessibility (ARIA) purposes.
   */
  ariaLabel?: string;
  /**
   * Specifies the style variant of the button.
   */
  variant?: Variant;
  effect?: Effects;
  /**
   * Additional properties for the button component.
   */
  [x: string]: any;
}
