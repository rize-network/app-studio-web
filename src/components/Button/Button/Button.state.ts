import React from 'react';

export const useButtonState = () => {
  const [isHovered, setIsHovered] = React.useState(false);

  return { isHovered, setIsHovered };
};
