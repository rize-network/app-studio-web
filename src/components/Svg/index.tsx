export * from './ArrowDown';
export * from './ArrowUp';
export * from './Check';
export * from './Close';
export * from './CloseEye';
export * from './DustBin';
export * from './Edit';
export * from './ExternalLink';
export * from './Indeterminate';
export * from './OpenEye';
export * from './Profile';
export * from './RightArrow';
export * from './Warning';
