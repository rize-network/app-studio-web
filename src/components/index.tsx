export * from './Button/Button';
export * from './Form/Checkbox/Checkbox';
export * from './Form/CountryPicker/CountryPicker';
export * from './Form/DatePicker/DatePicker';
export * from './Form/Password/Password';
export * from './Form/Select/Select';
export * from './Form/Switch/Switch';
export * from './Form/Switch/Switch';
export * from './Form/TextArea/TextArea';
export * from './Form/TextField/TextField';
export * from './Layout/Center/Center';
export * from './Formik';
export * from './Layout/Horizontal/Horizontal';
export * from './Layout/Vertical/Vertical';
export * from './Layout/View/View';
export * from './Link/Link';
export * from './Loader/Loader';
export * from './Modal/Modal';
export * from './Text/Text';
export * from './File/File';
export * from './Tabs/Tabs';
export * from './Alert/Alert';
export * from './AspectRatio/AspectRatio';
export * from './Avatar/Avatar';
export * from './Badge/Badge';
export * from './Toggle/Toggle';
export * from './ToggleGroup/ToggleGroup';
export * from './Message/Message';
