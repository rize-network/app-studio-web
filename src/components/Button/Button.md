### **Import**

```tsx static
import { Button } from 'app-studio';
```

### **Default**

```tsx
import { Center } from '../Layout/Center/Center';

<Center width="100%">
  <Button>Push Me</Button>
</Center>;
```

### **Disabled**

“isDisabled” makes the button unusable and un-clickable with a “disabled” look.

```tsx
import { Center } from '../Layout/Center/Center';
<Center width="100%">
  <Button isDisabled>Disabled</Button>
</Center>;
```

### **Size**

“size” alters the padding, text size, and border of the button. This attribute belongs to the "Sizes" type and has a default value of "md".

```jsx
import { Vertical } from '../Layout/Vertical/Vertical';

<Vertical gap={10}>
  {['xs', 'sm', 'md', 'lg', 'xl'].map((size, index) => (
    <Button key={index} size={size}>
      {size}
    </Button>
  ))}
  <Button isAuto>Auto Width</Button>
  <Button isFilled>Full Width</Button>
</Vertical>;
```

### **ColorScheme**

“colorScheme” allows you to modify the background color of the button. It accepts a value from the "ColorThemes" type, with the default color set to "theme.primary".

```tsx
import { Horizontal } from '../Layout/Horizontal/Horizontal';

<Horizontal justifyContent="space-evenly" gap={10}>
  {['theme.primary', 'theme.secondary', 'theme.warning', 'theme.success', 'theme.error'].map((color, index) => (
    <Button key={index} colorScheme={color} isAuto>
      {color}
    </Button>
  ))}
</Horizontal>;
```

### **Shadow**

"shadow" attribute applies a shadow effect to the button.

```tsx
import { Center } from '../Layout/Center/Center';
<Center width="100%">
  <Button shadow={{ boxShadow: 'rgb(249, 115, 22) 0px 4px 14px 0px' }}>Click Me</Button>
</Center>;
```

### **Loader**

"isLoader" and "loading" attributes display a loading animation, which indicates that an action is in progress.

```tsx
import { Loader } from '../Loader/Loader';
import { Text } from '../Text/Text';
import { Center } from '../Layout/Center/Center';

<Center width="100%">
  <Button isLoader>
    <Loader loaderColor="white" />
    <Text size="sm">Submitting</Text>
  </Button>
</Center>;
```

### **Variant**

"variant" allows for the customization of the button's styles. The attribute belongs to the "Variants" type and has a default variant of "filled".

```tsx
import { Horizontal } from '../Layout/Horizontal/Horizontal';

<Horizontal justifyContent="space-evenly" gap={10}>
  {['filled', 'outline', 'link', 'ghost'].map((variant, index) => (
    <Button key={index} variant={variant} colorScheme="theme.primary" isAuto>
      {variant}
    </Button>
  ))}
</Horizontal>;
```

### **Shape**

“shape” property changes the edges of the button. This property belongs to the "Shapes" type and has a default value of "rounded".

```tsx
import { Horizontal } from '../Layout/Horizontal/Horizontal';

<Horizontal justifyContent="space-evenly" wrap="nowrap" gap={10}>
  {['sharp', 'rounded', 'pillShaped'].map((border, index) => (
    <Button key={index} shape={border} isAuto>
      {border}
    </Button>
  ))}
</Horizontal>;
```

### **Icon**

“icon” provides the flexibility to incorporate any desired icon within the button.

```tsx
import { Center } from '../Layout/Center/Center';
import { DustBinSvg } from '../Svg/DustBin';

<Center justifyContent="space-evenly">
  <Button icon={<BsBucket size={24} />} height={48}>
    Delete
  </Button>
  <Button height={48} icon={<DustBinSvg size={24} />} shape={'pillShaped'} iconPosition="right" colorScheme="theme.secondary">
    Delete
  </Button>
  <Button icon={<DustBinSvg size={24} />} colorScheme="color.black" isIconRounded isAuto />
</Center>;
```

### Types

#### Sizes

```tsx static
type ButtonSizes = 'xs' | 'sm' | 'md' | 'lg' | 'xl';
```

#### Variants

```tsx static
type Variants = 'filled' | 'outline' | 'link' | 'ghost';
```

#### IconPositions

```tsx static
type IconPositions = 'left' | 'right';
```

#### Shapes

```tsx static
type Shapes = 'sharp' | 'rounded' | 'pillShaped';
```
