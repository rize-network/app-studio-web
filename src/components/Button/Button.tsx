import React from 'react';

import { ButtonProps } from './Button/Button.props';
import { useButtonState } from './Button/Button.state';
import ButtonView from './Button/Button.view';

const ButtonComponent: React.FC<ButtonProps> = (props: any) => {
  const { isHovered, setIsHovered } = useButtonState();
  const handleHover = () => setIsHovered(!isHovered);

  return (
    <ButtonView
      isHovered={isHovered}
      setIsHovered={setIsHovered}
      onMouseEnter={handleHover}
      onMouseLeave={handleHover}
      {...props}
    />
  );
};

/**
 * Buttons allow us to trigger an event or an action with a single click.
 */
export const Button = ButtonComponent;
