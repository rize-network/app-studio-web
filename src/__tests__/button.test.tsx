import React from 'react';
import renderer from 'react-test-renderer';
import { Button } from 'src/components/Button/Button';
import { Shape, Size } from 'src/components/Button/Button/Button.type';
import { DustBinSvg } from 'src/components/Svg';

import { cleanup, fireEvent, render, screen } from '@testing-library/react';

afterEach(() => {
  cleanup();
});

test('should render button component without crashing', () => {
  render(<Button />);
});

test('should render button component with correct text', () => {
  render(<Button>Button</Button>);
  const buttonElement = screen.getByRole('button');
  expect(buttonElement).toBeInTheDocument();
  expect(buttonElement).toHaveTextContent('Button');
});

test('should render button component with correct class name', () => {
  render(<Button className="button">Button</Button>);
  const buttonElement = screen.getByRole('button');
  expect(buttonElement).toHaveClass('button');
});

test('should render a button with fontSize equals to sm', () => {
  render(<Button size={'sm' as Size}>Button</Button>);
  const buttonElement = screen.getByRole('button');
  expect(buttonElement).toHaveStyle({ fontSize: 'sm' });
});

test('should render a button with a width equals to fit-content', () => {
  render(<Button isAuto>Button</Button>);
  const buttonElement = screen.getByRole('button');
  expect(buttonElement).toHaveStyle({ width: 'fit-content' });
});

test('should render a button with a width equals to 100%', () => {
  render(<Button isFilled>Button</Button>);
  const buttonElement = screen.getByRole('button');
  expect(buttonElement).toHaveStyle({ width: '100%' });
});

test('should render a button with backgroundColor theme.secondary', () => {
  render(<Button colorScheme="theme.secondary">Button</Button>);
  const buttonElement = screen.getByRole('button');
  expect(buttonElement).toHaveStyle({ backgroundColor: 'theme.secondary' });
});

test('button component calls onClick function when clicked', () => {
  const handleClick = jest.fn();
  render(<Button onClick={handleClick}>Button</Button>);
  const buttonElement = screen.getByRole('button');
  fireEvent.click(buttonElement);
  expect(handleClick).toHaveBeenCalledTimes(1);
});

test('Button with an icon on the left to match snapshot', () => {
  const tree = renderer
    .create(
      <Button icon={<DustBinSvg size={24} />} shape={'pillShaped' as Shape} colorScheme="theme.secondary">
        Delete
      </Button>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});

test('Button with an icon on the right to match snapshot', () => {
  const tree = renderer
    .create(
      <Button
        icon={<DustBinSvg size={24} />}
        shape={'pillShaped' as Shape}
        iconPosition="right"
        colorScheme="theme.secondary"
      >
        Delete
      </Button>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});

test('Button will a pillShaped shape to match snapshot', () => {
  const tree = renderer
    .create(
      <Button shape={'pillShaped' as Shape} isAuto>
        Pill Shape
      </Button>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
