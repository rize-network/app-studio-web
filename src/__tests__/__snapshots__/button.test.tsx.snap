// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button will a pillShaped shape to match snapshot 1`] = `
<button
  className="sc-bdfCDU eMuXTf"
  color="color.white"
  cursor="pointer"
  disabled={false}
  display="flex"
  filter="brightness(1)"
  
  fontWeight={600}
  letterSpacing={1.25}
  onClick={[Function]}
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
  role="button"
  transform=""
  width="fit-content"
>
  <div
    className="sc-bdfCDU dpBfNq"
    display="flex"
  >
    Pill Shape
  </div>
</button>
`;

exports[`Button with an icon on the left to match snapshot 1`] = `
<button
  className="sc-bdfCDU eSlpEH"
  color="color.white"
  cursor="pointer"
  disabled={false}
  display="flex"
  filter="brightness(1)"
  
  fontWeight={600}
  letterSpacing={1.25}
  onClick={[Function]}
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
  role="button"
  transform=""
  width={144}
>
  <div
    className="sc-bdfCDU dpBfNq"
    display="flex"
  >
    <div
      className="sc-bdfCDU dMZcpT"
      display="flex"
      height="24px"
      width="24px"
    >
      <svg
        fill="#ffffff"
        height="24px"
        id="Capa_1"
        version="1.1"
        viewBox="0 0 197.516 197.516"
        width="24px"
        xmlns="http://www.w3.org/2000/svg"
      >
        <g
          id="SVGRepo_bgCarrier"
          strokeWidth="0"
        />
        <g
          id="SVGRepo_tracerCarrier"
          strokeLinecap="round"
          strokeLinejoin="round"
        />
        <g
          id="SVGRepo_iconCarrier"
        >
          <path
            d="M68.758,170.083V72.649h15v97.434H68.758z M128.758,72.649h-15v97.434h15V72.649z M140.539,0v12.631h34.885v47.746h-10.525 v137.139H32.617V60.377H22.092V12.631h34.883V0H140.539z M149.898,60.377H47.617v122.139h102.281V60.377z M125.539,27.631V15H71.975 v12.631H37.092v17.585h123.332V27.631H125.539z"
          />
        </g>
      </svg>
    </div>
    Delete
  </div>
</button>
`;

exports[`Button with an icon on the right to match snapshot 1`] = `
<button
  className="sc-bdfCDU eSlpEH"
  color="color.white"
  cursor="pointer"
  disabled={false}
  display="flex"
  filter="brightness(1)"
  
  fontWeight={600}
  letterSpacing={1.25}
  onClick={[Function]}
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
  role="button"
  transform=""
  width={144}
>
  <div
    className="sc-bdfCDU dpBfNq"
    display="flex"
  >
    Delete
    <div
      className="sc-bdfCDU dMZcpT"
      display="flex"
      height="24px"
      width="24px"
    >
      <svg
        fill="#ffffff"
        height="24px"
        id="Capa_1"
        version="1.1"
        viewBox="0 0 197.516 197.516"
        width="24px"
        xmlns="http://www.w3.org/2000/svg"
      >
        <g
          id="SVGRepo_bgCarrier"
          strokeWidth="0"
        />
        <g
          id="SVGRepo_tracerCarrier"
          strokeLinecap="round"
          strokeLinejoin="round"
        />
        <g
          id="SVGRepo_iconCarrier"
        >
          <path
            d="M68.758,170.083V72.649h15v97.434H68.758z M128.758,72.649h-15v97.434h15V72.649z M140.539,0v12.631h34.885v47.746h-10.525 v137.139H32.617V60.377H22.092V12.631h34.883V0H140.539z M149.898,60.377H47.617v122.139h102.281V60.377z M125.539,27.631V15H71.975 v12.631H37.092v17.585h123.332V27.631H125.539z"
          />
        </g>
      </svg>
    </div>
  </div>
</button>
`;
